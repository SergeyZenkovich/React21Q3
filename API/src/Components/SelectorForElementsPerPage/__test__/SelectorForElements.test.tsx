import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import styled from "styled-components";
import "jest-styled-components";
import {Provider} from "react-redux";
import Selector from "../SelectorForElementsPerPage";
import store from "../../../redux/store";

const testObj = {
  elementsOnPage: 10
};

it("renders Selector correctly", () => {
  const container = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Selector {...testObj}></Selector>
    </Provider>,
    container
  );

  expect(container.querySelector("h5").textContent).toBe("Elemetns per page:");
});

describe("Selector Component", () => {
  const setFunction = jest.fn();
  let container;
  beforeEach(() => {
    container = renderer.create(
      <Provider store={store}>
        <Selector {...testObj}></Selector>
      </Provider>
    );
  });
  it("matches shapshot", () => {
    const tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
