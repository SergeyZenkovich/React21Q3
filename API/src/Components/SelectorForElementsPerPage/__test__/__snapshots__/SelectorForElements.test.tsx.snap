// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Component matches shapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 h5 {
  margin: 0;
  padding: 0;
  font-family: "CardTextFont";
  font-size: 21px;
}

.c0 select {
  margin: 0;
  padding: 0 10px;
  height: 40px;
  font-family: "CardTextFont";
  font-size: 20px;
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <h5>
    Elemetns per page:
  </h5>
  <select
    id="resultsCount"
    name="resultsCount"
    onChange={[Function]}
    value={10}
  >
    <option
      value="2"
    >
      2
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
</div>
`;
